server/models/User.js

```diff
@@ -3,14 +3,32 @@ const Schema = mongoose.Schema;
 const bcrypt = require("bcrypt");
 
 const userSchema = new Schema({
-  username: { type: String },
-  email: { type: String },
+  username: {
+    type: String,
+    trim: true,
+    unique: true,
+    required: [true, "A username is required."],
+    minLength: [2, "Username must be at least 2 characters long."],
+    maxLength: [32, "Username cannot exceed 32 characters"],
+  },
+  email: {
+    type: String,
+    required: [true, "An email is required."],
+    trim: true,
+    lowercase: true,
+    unique: true,
+    match: [
+      /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/,
+      "Please fill a valid email address",
+    ],
+  },
   password: {
     type: String,
+    required: [true, "A password is required"],
+    minLength: [6, "Password must be at least 6 characters long"],
```

Comment:
if we have a max on the front end for password length, we should on the back end as well

Reply:
Done and changed.